### PR TITLE
Fix animation example

### PR DESCRIPTION
**Pull request by kojix2**
This pull request fixes the problem of showing "1000" in animation example.

image

The plot contains the ANSI escape code.
Even if the length of the string printed on the terminal does not change, the actual length of the string may change.
If the string is too short to overwrite all the characters, you will see "1000" left at the end.

### CODE REVIEW

**Comment by kojix2** [comment]
In most cases, SIGINT will be sent at the pink point. However, in some cases, SIGINT is sent during the execution of the green line.

If the SIGINT is sent during the execution of the green line, the code in the yellow area will not be executed. If SIGINT is sent at the pink point, the code in the yellow area is already executed.

Therefore, there are two different possible positions for the cursor when SIGINT is sent and finished. This can be confirmed by actually trying Ctrl+C many times.

image

The second commit is intended to change the code so that SIGINT behaves the same no matter when it is sent.

File: example/animation.rb [hunk 1]
```diff
@@ -22,13 +22,14 @@ def out.tty?; true; end
   lines.each do |line|
     $stdout.print "\r#{line}"
   end
+  $stdout.print "\e[0J"
```
Comment (mrkn) [inline example/animation.rb:25]: @kojix2 Why is this necessary?

Comment (kojix2) [reply example/animation.rb:25]: It is not necessary, but it will ensure that there are no more writes.

Comment (kojix2) [reply example/animation.rb:25]: For example, if you hit the keyboard repeatedly during execution, the characters will be erased.

Comment (mrkn) [reply example/animation.rb:25]: I got it. Thank you!

File: example/animation.rb [hunk 2]
```diff
@@ -25,11 +25,13 @@ def out.tty?; true; end
...
-  $stdout.print "\e[#{n}F"
-  shift = (shift + M) % N
-
   sleep 0.2
+
+  if continue
+    n = lines.count
+    $stdout.print "\e[#{n}F"
+    shift = (shift + M) % N
+  end
```
Comment (kojix2) [inline example/animation.rb]: I see. I should have understood your idea of not erasing the plot on exit.
Sorry.

**Comment by kojix2** [comment]
You were right in your idea.

It's much better to leave the plot in the terminal when the animation and programming is finished.

I realized that yesterday. Then I remembered this pull request. 
I will change my mind and also change the behavior of YouPlot.

Thanks.
